pkg/providers/instancetype/suite_test.go

```diff
@@ -1186,6 +1185,22 @@ var _ = Describe("InstanceType Provider", func() {
 			vm := azureEnvNonZonal.VirtualMachinesAPI.VirtualMachineCreateOrUpdateBehavior.CalledWithInput.Pop().VM
 			Expect(vm.Zones).To(BeEmpty())
 		})
+		It("should support provisioning non-zonal instance types in zonal regions", func() {
+			nonZonalInstanceType := "Standard_NC6s_v3"
+			ExpectApplied(ctx, env.Client, nodePool, nodeClass)
+
+			pod := coretest.UnschedulablePod(coretest.PodOptions{
+				NodeSelector: map[string]string{v1.LabelInstanceTypeStable: nonZonalInstanceType},
+			})
+			ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, coreProvisioner, pod)
+
+			node := ExpectScheduled(ctx, env.Client, pod)
+			Expect(node.Labels["node.kubernetes.io/instance-type"]).To(Equal(nonZonalInstanceType))
```

Comment:
since we already are popping the vm below, could we also check the VMSize in properties.HardwareProfile.VMSize?

Reply:
We could (even though, all things being equal, it is nice to use provider-independent checks), but I don't think we need either check for instance type, since many other tests verify instance type constraints are respected. What we _should_ check here instead is the zone label `v1alpha2.AlternativeLabelTopologyZone`).

Reply:
The above comment was only about checking instance type being unnecessary, `Expect(vm.Zones).To(BeEmpty())` is very much appropriate, can we restore it?